jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ContainerConfiguration.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle;
+
+import com.google.cloud.tools.jib.image.ImageFormat;
+import com.google.cloud.tools.jib.image.json.BuildableManifestTemplate;
+import com.google.common.base.Preconditions;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Optional;
+
+/**
+ * A bean that configures properties of the container run from the image. This is configurable with
+ * Groovy closures and can be validated when used as a task input.
+ */
+public class ContainerConfiguration {
```

Comment:
Maybe `ContainerParameters` so that it's not confused with the container configuration that is part of the image format?